src/main/java/co/rsk/federate/signing/hsm/message/ReleaseCreationInformationGetter.java

```diff
@@ -144,24 +145,30 @@ private ReleaseCreationInformation searchEventInFollowingBlocks(
             blockNumber,
             block.getTransactionsList().size()
         );
-        for (Transaction transaction : block.getTransactionsList()) {
-            TransactionInfo transactionInfo = receiptStore.getInMainChain(transaction.getHash().getBytes(), blockStore).orElse(null);
-            TransactionReceipt transactionReceipt = transactionInfo.getReceipt();
-            transactionReceipt.setTransaction(transaction);
-            Optional<ReleaseCreationInformation> optionalReleaseCreationInformation =
-                getInformationFromEvent(block, transactionReceipt, btcTransaction, rskTxHash, informingRskTxHash);
-            if (optionalReleaseCreationInformation.isPresent()) {
-                return optionalReleaseCreationInformation.get();
+        for (Transaction tx : block.getTransactionsList()) {
+            TransactionReceipt txReceipt = receiptStore.getInMainChain(tx.getHash().getBytes(), blockStore)
+                .map(TransactionInfo::getReceipt)
+                .orElseThrow(() -> new HSMReleaseCreationInformationException(
+                    String.format("[searchEventInFollowingBlocks] Transaction hash [%s] should exist", tx.getHash())));
```

Comment:
```suggestion
            TransactionReceipt txReceipt = receiptStore.getInMainChain(tx.getHash().getBytes(), blockStore)
                .map(TransactionInfo::getReceipt)
                .orElseThrow(() -> new HSMReleaseCreationInformationException(
                        String.format("[searchEventInFollowingBlocks] Rsk transaction hash [%s] should exist", tx.getHash()
                        )
                    )
                );
```

Reply:
Updated the log statement, as well in other places in the same method with similar issues.